### PR TITLE
interfaces: fix default content attribute value (2.23)

### PR DESCRIPTION
**Pull request by mvo5**
Our documentation and tutorials state that the default value
for the "content" attribute in the content interface is the
plug/slot name if not specified otherwise. This branch ensures
this is the case.

### CODE REVIEW

File: tests/lib/snaps/test-snapd-content-slot-empty-content-attr/meta/snap.yaml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+name: test-snapd-content-slot-empty-content-attr
+version: 1.0
+slots:
+    shared-content:
+        interface: content
+        read:
+            - /
```
Comment (zyga) [inline tests/lib/snaps/test-snapd-content-slot-empty-content-attr/meta/snap.yaml]: This should say `$SNAP/` instead. The old syntax is legacy and we should not advertise it more.

File: tests/main/interfaces-content-empty-content-attr/task.yaml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+    echo "And a snap declaring a content sharing plug is installed"
+    snap install --edge test-snapd-content-plug-empty-content-attr
+
+execute: |
+    CONNECTED_PATTERN="test-snapd-content-slot-empty-content-attr:shared-content +test-snapd-content-plug-empty-content-attr"
+    DISCONNECTED_PATTERN="(?s).*?test-snapd-content-slot-empty-content-attr:shared-content +-.*?- +test-snapd-content-plug-empty-content-attr"
+
+    echo "Then the snap is listed as connected"
+    snap interfaces | grep -Pzq "$CONNECTED_PATTERN"
```
Comment (zyga) [inline tests/main/interfaces-content-empty-content-attr/task.yaml]: Could we use MATCH instead?

File: tests/main/interfaces-content-empty-content-attr/task.yaml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    [ $(find /var/lib/snapd/mount -type f -name "*.fstab" | wc -l) -gt 0 ]
+
+    echo "And we can use the shared content"
+    test-snapd-content-plug-empty-content-attr.content-plug | grep "Some shared content"
+
+    echo "============================================"
+
+    echo "When the plug is disconnected"
+    snap disconnect test-snapd-content-plug-empty-content-attr:shared-content test-snapd-content-slot-empty-content-attr:shared-content
+    snap interfaces | grep -Pzq "$DISCONNECTED_PATTERN"
```
Comment (zyga) [inline tests/main/interfaces-content-empty-content-attr/task.yaml]: MATCH

File: tests/main/interfaces-content-empty-content-attr/task.yaml [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+    echo "Then the fstab files are removed"
+    [ $(find /var/lib/snapd/mount -type f -name "*.fstab" | wc -l) -eq 0 ]
+
+    echo "When the plug is reconnected"
+    snap connect test-snapd-content-plug-empty-content-attr:shared-content test-snapd-content-slot-empty-content-attr:shared-content
+    snap interfaces | grep -Pzq "$CONNECTED_PATTERN"
```
Comment (zyga) [inline tests/main/interfaces-content-empty-content-attr/task.yaml]: MATCH

**Comment by mvo5** [comment]
@zyga thanks a lot! I addressed the issues now.

File: tests/lib/snaps/test-snapd-content-slot-empty-content-attr/meta/snap.yaml (hunk 1 above)
Comment (mvo5) [inline tests/lib/snaps/test-snapd-content-slot-empty-content-attr/meta/snap.yaml]: I will do a separate branch for this for master. When fixing it in 2.23 I got some strange errors and I do not want to hold up 2.23.1 futher.

File: tests/main/interfaces-content-empty-content-attr/task.yaml (hunk 2 above)
Comment (mvo5) [inline tests/main/interfaces-content-empty-content-attr/task.yaml]: Unfortunately the options are not compatible, match does not do it here. Should be something we chase in master too, I agree that MATCH is much nicer here.

**Pull request merged**